src/interfaces/IAppliance.js

```diff
@@ -0,0 +1,95 @@
+import {
+	InterfaceInstantiationError,
+	NotImplementedError,
+} from '../errors'
+
+class IAppliance {
+	/**
+	 * Constructor for a new appliance
+	 * @param  {Object} overrideSettings Values to override default appliance settings.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	constructor(overrideSettings = {}) {
+		if (this.constructor === IAppliance) {
+			throw new InterfaceInstantiationError(this.constructor.name)
+		}
+	}
+
+	/**
+	 * Getter for the list of data types accepted by the appliance.
+	 *
+	 * @return {String[]} The list of data types accepted by the appliance.
+	 */
+	getInputTypes = () => {
+		throw new NotImplementedError('getInputTypes')
+	}
+
+	/**
+	 * Getter for the list of data types produced by the appliance.
+	 *
+	 * @return {String[]} The list of data types produced by the appliance.
+	 */
+	getOutputTypes = () => {
+		throw new NotImplementedError('getOutputTypes')
+	}
+
+	/**
+	 * Asserts that a given payload is actually a payload.
+   *
+	 * @param  {Payload} payload The payload that we want to validate.
+	 * @return {Boolean}         The result of the validation check.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	isValidPayload = async (payload) => {
+		throw new NotImplementedError('isValidPayload')
+	}
+
+	/**
+	 * Invokes the appliance on any unprocessed data in the appliance buffer.
+	 *
+	 * @return {Boolean} Whether or not the appliance had enough data to work with.
```

Comment:
Does the boolean response specifically mean whether it "had enough data to work with"? Or are there other reasons it could return true/false? Just wondering if this should be genericized a bit more, to something like "whether the appliance was able to begin working" or something.

Reply:
Good question.

I think... yes?  And maybe we use errors if there's some kind of issue with the input?